### PR TITLE
Add bridge vlan-filtering support

### PR DESCRIPTION
**Pull request by bengal**
https://bugzilla.redhat.com/show_bug.cgi?id=1652910

### CODE REVIEW

File: src/devices/nm-device-bridge.c [hunk 1]
```diff
@@ -396,22 +387,101 @@ master_update_slave_connection (NMDevice *device,
...
+	self->vlan_configured = TRUE;
+
+	/* Filtering must be disabled to change the default PVID */
+	r = nm_platform_sysctl_master_set_option (plat, ifindex, "vlan_filtering", "0");
+	if (!r)
+		return FALSE;
+
+	/* Clear the default PVID so that we later can force the re-creation of
+	 * default PVID VLANs by writing the option again. */
+	nm_platform_sysctl_master_set_option (plat, ifindex, "default_pvid", "0");
```
Comment (fgiudici) [inline src/devices/nm-device-bridge.c]: **r =** nm_platform_sysctl_master_set_option

Comment (bengal) [reply src/devices/nm-device-bridge.c]: Fixed, thanks (removed the 'r' variable and checked the function return value directly).

File: clients/common/nm-meta-setting-desc.c [hunk 2]
```diff
@@ -3482,6 +3512,58 @@ _objlist_set_fcn_tc_config_qdiscs (NMSetting *setting,
...
+		nm_setting_bridge_add_vlan (NM_SETTING_BRIDGE (setting), vlan);
+	else {
+		nm_setting_bridge_remove_vlan_by_vid (NM_SETTING_BRIDGE (setting),
+		                                      nm_bridge_vlan_get_vid (vlan));
+	}
+	return TRUE;
+}
+
+static gboolean
+_objlist_set_fcn_bridge_port_vlans (NMSetting *setting,
```
[the remark this answers is not in the document]
Comment (bengal) [reply clients/common/nm-meta-setting-desc.c]: done

File: libnm-core/nm-keyfile.c [hunk 3]
```diff
@@ -1518,6 +1518,49 @@ team_config_parser (KeyfileReaderInfo *info, NMSetting *setting, const char *key
...
+		NMBridgeVlan *vlan;
+		const char *index;
+		gs_free char *vlan_rest = NULL;
+		gs_free char *vlan_str = NULL;
+		gs_free_error GError *err = NULL;
+
+		if (!g_str_has_prefix (keys[i], "vlan."))
+			continue;
+
+		index = keys[i] + sizeof ("vlan.") - 1;
```
[the remark this answers is not in the document]
Comment (bengal) [reply libnm-core/nm-keyfile.c]: fixed

File: libnm-core/nm-keyfile.c [hunk 4]
```diff
@@ -1842,6 +1885,37 @@ route_writer (KeyfileWriterInfo *info,
...
+		return;
+
+	for (i = 0; i < array->len; i++) {
+		NMBridgeVlan *vlan = array->pdata[i];
+		GString *key_name = g_string_sized_new (16);
+		GString *value_str = g_string_sized_new (20);
+
+		g_string_append (key_name, "vlan.");
+		g_string_append_printf (key_name, "%u", nm_bridge_vlan_get_vid (vlan));
+		_nm_bridge_vlan_str_append_rest (vlan, value_str, FALSE);
```
[the remark this answers is not in the document]
Comment (bengal) [reply libnm-core/nm-keyfile.c]: ok

File: clients/common/nm-meta-setting-desc.c [hunk 5]
```diff
@@ -4814,6 +4896,17 @@ static const NMMetaPropertyInfo *const property_infos_BRIDGE[] = {
...
 		.property_type =                &_pt_gobject_int,
 	),
+	PROPERTY_INFO_WITH_DESC (NM_SETTING_BRIDGE_VLANS,
+		.property_type =                &_pt_objlist,
+		.property_typ_data = DEFINE_PROPERTY_TYP_DATA (
+			PROPERTY_TYP_DATA_SUBTYPE (objlist,
+				.get_num_fcn =          OBJLIST_GET_NUM_FCN         (NMSettingBridge, nm_setting_bridge_get_num_vlans),
+				.clear_all_fcn =        OBJLIST_CLEAR_ALL_FCN       (NMSettingBridge, nm_setting_bridge_clear_vlans),
+				.obj_to_str_fcn =       _objlist_obj_to_str_fcn_bridge_vlans,
+				.set_fcn =              _objlist_set_fcn_bridge_vlans,
```
[the remark this answers is not in the document]
Comment (bengal) [reply clients/common/nm-meta-setting-desc.c]: Will do once routing-rule-pt2 is merged to master.

**Comment by bengal** [comment]
Merged as fc4aedc6f37842d6572e7db6c97e0fe1a392ba0d.

**Pull request closed without merging**